core/src/test/java/kafka/server/AssignmentsManagerTest.java

```diff
@@ -72,6 +75,29 @@ void tearDown() throws InterruptedException {
         manager.close();
     }
 
+    AssignReplicasToDirsRequestData normalize(AssignReplicasToDirsRequestData request) {
```

Comment:
@soarez Why are we creating a copy instead of simply sorting in place?

Reply:
@divijvaidya that sounds like a good idea to me. This list sorting aspect probably applies to other schemas too. Maybe we could extend the message generator framework to allow specifying whether the order matters in `FieldType.ArrayType`. At the moment I'm focussed on making as much progress on JBOD as possible for 3.7, but I'm happy to come back and tackle this later if no else picks it up in the meantime.

@ijuma because this is a test, so reasons not to copy don't really take place such as extra time, extra memory footprint; but the reasons to copy do: minimizes side effects from the caller perspective, especially as this method is exclusively used as a utility to `assertRequestEquals` — it would be surprising to find that a method that verifies equality is also modifying its arguments.

Reply:
Thanks for the response @soarez! 

I agree with not modifying the objects for assertion. Copying should be ok for tests.

Reply:
This approach results in more maintenance, any change to this schema will require changing this method. I don't agree it's better.

Reply:
We could actually meet in the middle. How about duplicating the structure with `duplicate()` and sort in place? I think that this would address both issues.

Reply:
That's fine. My issue is not with the copy, it's all the code required to manually set all the fields.